### DISCUSSION TITLE
Identical calls to nominatim.osm.org return different addresses

### DISCUSSION THREAD

**Discussion by jilkoval** [osm-search/Nominatim#3418] [General]
Hey, 
Identical repeated calls to Nominatim using geopy return a multiple location addresses while the coordinates stay the same. I’m wondering if this is expected behavior? Thank you.

Specifications: Python 3.11.5, geopy 2.4.1

Example:
```python
import time
from geopy.geocoders import Nominatim

query = “University of Auckland”
app = Nominatim(user_agent=“my_app”)

for i in range(5):
    result = app.geocode(query=query, exactly_one=False, timeout=None)
    print(result[0].address)
    print(result[0].latitude, result[0].longitude)
    time.sleep(150)
```

Output:
```
Waipapa Taumata Rau / University of Auckland, Evelyn Street, Kingsland, Albert-Eden, Auckland, 1002, New Zealand / Aotearoa
-36.8660955 174.77373309361866
Waipapa Taumata Rau / University of Auckland, Evelyn Street, Kingsland, Albert-Eden, Auckland, 1002, New Zealand / Aotearoa
-36.8660955 174.77373309361866
Waipapa Taumata Rau / University of Auckland, Evelyn Street, Kingsland, Albert-Eden, Auckland, 1002, New Zealand / Aotearoa
-36.8660955 174.77373309361866
Waipapa Taumata Rau / University of Auckland, Victoria Street West Cycleway, Victoria Quarter, Wynyard Quarter, Auckland, Waitematā, Auckland, 1001, New Zealand / Aotearoa
-36.8660955 174.77373309361866
Waipapa Taumata Rau / University of Auckland, Victoria Street West Cycleway, Victoria Quarter, Wynyard Quarter, Auckland, Waitematā, Auckland, 1001, New Zealand / Aotearoa
-36.8660955 174.77373309361866
```

**Comment by mtmail** [comment]
There are currently 4 servers behind the nominatim.openstreetmap.org URL. It's possible one of them is not in sync with the others.

**Comment by mtmail** [comment]
My local planet Nominatim installation(s) return yet another address for that query

`Waipapa Taumata Rau / University of Auckland, Bankside Street, Britomart, City Centre, Auckland, Waitematā, Auckland, 1010, New Zealand / Aotearoa`

Do you know what the best result would be for https://www.openstreetmap.org/relation/10726338 ? On their website I only see a postbox address, not street address https://www.auckland.ac.nz/en/about-us/connect-with-us/contact-us.html

**Comment by jilkoval** [comment]
I also got `University of Auckland,"Waipapa Taumata Rau / University of Auckland, Pukekawa Centennial Glade Path, Learning Quarter, Grafton, Auckland, Waitematā, Auckland, 1053, New Zealand / Aotearoa` at some point. 

> Do you know what the best result would be for https://www.openstreetmap.org/relation/10726338 ?

The relation covers several campuses so it's hard to pick a single representative address. And it's probably not any of those returned by Nominatim at this point. I'd suggest _The Clock Tower_ `22 Princes Street, Auckland, 1010, New Zealand / Aotearoa` where is the administrative headquarters for the University.

**Reply by mtmail** [reply]
'22 Princes Street' didn't exist yet in OSM data, I added that in front of the ClockTower building. I see the student information centre is located in that building.

**Comment by lonvia** [comment]
Couple of comments on this:

* The relation is tagged wrongly as `type=boundary` while it really should be `type=site`.
* There is a long-standing issue that we need a new level in the address hierarchy for site objects like this. They probably shouldn't get a street address at all. #463 covers support for site relations.
* #3390 already somewhat mitigates the problem for reverse search, as you should usually be getting a more specific POI than the site, if one is available.

That said, you can't expect reproducable results when using nominatim.openstreetmap.org. Not only are there 4 different servers, they also get updated every minute with new edits from OpenStreetMap, So that results change.

Moving this to a discussion.

**Reply by mtmail** [reply]
There's already a `type=site` https://www.openstreetmap.org/relation/14960032 covering much larger area. Could https://www.openstreetmap.org/relation/10726338 become a `type=multipolygon` with `addr:` tags? That's how it was done in Cologne, Germany https://www.openstreetmap.org/relation/8727202

**Reply by lonvia** [reply]
A university campus doesn't really have a street address. Usually it covers a large area that contains multiple real street addresses. Thus, it shouldn't have any `addr:`* tags at all. According to the [university's website](https://www.auckland.ac.nz/en/about-us/connect-with-us/contact-us.html), their main address is a postbox. If you want to capture that, the better choice would be `contact:*` tags.

The trouble with Nominatim is that it handles a university campus the same way as a building or POI and tries to assign a street address anyway. That's a software issue, but one which results from a tagging issue: the `amenity=university` tag currently is used sometimes for the site (a collection relation with all the buildings of the university), sometimes the campus (a huge area) and sometimes for single university buildings. I've considered handling university the same as landuse (see #3308), it might mean that we err on the less annoying side.

Regarding multipolygon vs. site: if you look into the Cologne university relation, you will find that it is also in fact mostly a collection of buildings. This is not what a multipolygon is meant for. They should only be used to describe complex areas. Marking the difference between area geometry and collection of objects can be quite important for a data user.

**Reply by mtmail** [reply]
I received a changeset comment https://www.openstreetmap.org/changeset/151130960 that type=site breaks rendering. (I don't see any rendering issues when I browse the map)

**Reply by mtmail** [reply]
![image](https://github.com/osm-search/Nominatim/assets/3727288/bfdc36b5-0fe9-40d5-9c4f-c52737061b2d)
Why can't the complex world fit into a simply data schema? :)

**Reply by lonvia** [reply]
Well, I can't really argue there because we don't have a good tagging schema to offer that works and is understood by data users. Change it back but use at least `type=multipolygon`? `type=boundary` was clearly not a good fit and just works by accident for most renderers. And maybe add a `landuse=education` for good measure.
